morpheus/_lib/tests/CMakeLists.txt

```diff
@@ -37,4 +37,23 @@ add_test(
   COMMAND $<TARGET_FILE:test_libmorpheus>
 )
 
+if(MORPHEUS_PYTHON_INPLACE_BUILD)
+  set(test_rpath "${MORPHEUS_LIB_ROOT}")
+else()
+  set(test_rpath "${Python3_SITELIB}/${PROJECT_NAME}/_lib")
+endif()
```

Comment:
This makes me nervous because it relies on several assumptions being true. 1) If `MORPHEUS_PYTHON_INPLACE_BUILD=OFF`, that `pip install` has been run (you get a load error if it has not) 2) If `MORPHEUS_PYTHON_INPLACE_BUILD=ON`, that the `libmorpheus_util.so` still exists in the source.

In general, I think its bad practice to change the RPATH based on `MORPHEUS_PYTHON_INPLACE_BUILD`. You just have to make too many assumptions between the build and source folders. All this flag is supposed to do is copy the binaries back to the source, unchanged.

Instead, I think we can design this in a way where it entirely works off of relative paths and `$ORIGIN`. That way, it doesn't matter where the tests are copied to, they will work. To do this, instead of doing the following steps:

1. Post build
   2. Install morpheus into conda environment
   3. Tar up conda environment
   4. Tar up C++ tests
5. Pre tests
   6. Untar conda environment
   7. Untar C++ tests
   8. Run tests

We should:

1. Post build
   2. Tar up conda environment (without Morpheus installed)
   3. Tar up entire `build/wheel` directory (this should include the C++ tests)
5. Pre tests
   6. Untar conda environment
   7. Untar `build/wheel` directory
   8. Pip install from `build/wheel` to put morpheus in conda environment
   9. Run C++ tests from `build/wheel` directory

This way, no matter what goes into the `build/wheel` directory, it will make it to the test environment. We eliminate the need to set the RPATHs depending on the inplace build flag. And as long as the `build/wheel` directory is copied as a unit, things will work. You might need to see what the optimal